channeldb/channel.go

```diff
@@ -3769,3 +3742,118 @@ func DKeyLocator(r io.Reader, val interface{}, buf *[8]byte, l uint64) error {
 func MakeKeyLocRecord(typ tlv.Type, keyLoc *keychain.KeyLocator) tlv.Record {
 	return tlv.MakeStaticRecord(typ, keyLoc, 8, EKeyLocator, DKeyLocator)
 }
+
+// serializeRevocationLogEntry takes several fields from a ChannelCommitment
+// to construct a revocation log entry that's written to the io.Writer.
+func serializeRevocationLogEntry(w io.Writer, c *ChannelCommitment) error {
+	if err := WriteElements(w,
+		c.LocalBalance, c.RemoteBalance, c.FeePerKw, c.CommitTx,
```

Comment:
The commitment transaction is still being serialized here though?

Reply:
Do we actually still need it? Given that in the event of a breach, we get notified that a spend happens, and that spend notification will carry the commitment transaction along with it. When we go to write that to the retribution store, then we can store the final breached commitment transaction only once.

Reply:
> The commitment transaction is still being serialized here though?

This is later removed. I took gradual step-by-step commits here to show the changes, just wanna be careful.

> Do we actually still need it?

Very good question! I have the same thought because the breach tx would be detected and [parsed here](https://github.com/lightningnetwork/lnd/blob/master/contractcourt/chain_watcher.go#L563). And right now we do need to commitment tx stored in revocation to perform [this check](https://github.com/lightningnetwork/lnd/blob/master/contractcourt/chain_watcher.go#L804). I think we could even remote this check completely? Even for the `LocalBalance` and `RemoteBalance` I think we can find them in the breach tx's outputs, so we don't even need to save them at all?
Seems like a bold move, need more check.